File: DSA/VaniAgrawal_590015846/Day_11_Question_2.py
# Find the longest palindromic substring in a given string
s = input("enter string :")

length = 0
for _ in s:
    length += 1

start = 0
max_len = 1

i = 0
while i < length:
    
    left = i
    right = i
    while left >= 0 and right < length and s[left] == s[right]:
        curr_len = right - left + 1
        if curr_len > max_len:
            start = left
            max_len = curr_len
        left -= 1
        right += 1

    
    left = i
    right = i + 1
    while left >= 0 and right < length and s[left] == s[right]:
        curr_len = right - left + 1
        if curr_len > max_len:
            start = left
            max_len = curr_len
        left -= 1
        right += 1

    i += 1

j = 0
while j < length:
    if j >= start and j < start + max_len:
        print(s[j], end='')
    j += 1

File: DSA/RATNA_KHANNA_590013141/DAY_11_QUESTION_2.c
#include <stdio.h>
#include <string.h>

int main() {
    char str[1000];
    printf("Enter a string: ");
    scanf("%s", str);

    int len = strlen(str);
    int start = 0, maxLength = 1;

    for (int i = 0; i < len; i++) {
        int left = i, right = i;
        while (left >= 0 && right < len && str[left] == str[right]) {
            if ((right - left + 1) > maxLength) {
                start = left;
                maxLength = right - left + 1;
            }
            left--;
            right++;
        }

        left = i;
        right = i + 1;
        while (left >= 0 && right < len && str[left] == str[right]) {
            if ((right - left + 1) > maxLength) {
                start = left;
                maxLength = right - left + 1;
            }
            left--;
            right++;
        }
    }

    printf("Longest Palindromic Substring: ");
    for (int i = start; i < start + maxLength; i++) {
        printf("%c", str[i]);
    }
    printf("\n");

    return 0;
}

